### PR TITLE
Playout priors

### PR DESCRIPTION
**Pull request by ujh**
I've now added the priors to the playouts. Unfortunately this makes the `engine::uct::node::test::full_uct_cycle_13x13` benchmark 4 times slower. :anguished: Interestingly though it still increases the playout strength on 9x9 to 72% +/- 2.1% (up from 68% +/- 3.3) which is still better but also within the error of the previous version. :sob: And on 13x13 we're down to 9.5% +/- 1.4 (previously 10% +/- 2.1).

I'm currently running the benchmarks with both pattern priors turned on. Maybe that will increase the strength a bit.

In summary: The playout prior is useful, but at this point it is just good enough to make up the speed loss it incurs. If we can manage to speed it up some more we might actually see some speed increase on bigger boards.

### CODE REVIEW

**Comment by ujh** [comment]
And I forgot to mention: I removed the possibility to select different playout kinds. The configuration for this stuff now lives in `Config` like everything else.

**Comment by ujh** [comment]
I also ran the code with the pattern prior turned on and here the strength on 9x9 is 67% +/- 3.3 (the same as without any patterns) and on 13x13 it's 11% +/- 2.2 (same as 0.2.4).

So I'll just merge it like it is (with 3x3 patterns in the playouts but none in the tree) and rewrite the `Config` to support different default per board size so that I can turn on the patterns in the tree for 13x13 (and bigger).
